drivers/alsamidi/midi_driver_alsamidi.cpp

```diff
@@ -84,8 +102,23 @@ void MIDIDriverALSAMidi::thread_func(void *p_udata) {
 						}
 						expected_size = get_message_size(byte);
 					}
-
-					if (bytes < 256) {
+					
+					if (expected_size == -1) {
+						// SysEx can be >256 bytes, and their contents will be ignored,
+						// so we read and discard until a SysEx end
+						buffer[0] = byte;
+						do {
+							ret = snd_rawmidi_read(midi_in, &byte, 1);
+						} while (byte != 0xF7 && ret >= 0);
+						if (ret < 0) {
+							if (ret != -EAGAIN) {
+								ERR_PRINTS("snd_rawmidi_read error: " + String(snd_strerror(ret)));
+							}
+						} else {
+							md->receive_input_packet(timestamp, buffer, 1);
+							bytes = 0;
+						}
+					} if (bytes < 256) {
```

Comment:
This is misleading, is it intended to be an `else if`? If not, it should be on its own line, not on the closing bracket of the previous `if`.